sdk/communication/azure-communication-rooms/src/test/java/com/azure/communication/rooms/RoomsAsyncClientTests.java

```diff
@@ -41,234 +38,371 @@ protected void afterTest() {
-            }).verifyComplete();
+                .assertNext(result4 -> {
+                    assertHappyPath(result4, 200);
+                }).verifyComplete();
 
-        Mono<Response<Void>> response5 =  roomsAsyncClient.deleteRoomWithResponse(roomId);
+        Mono<Response<Void>> response5 = roomsAsyncClient.deleteRoomWithResponse(roomId);
         StepVerifier.create(response5)
-        .assertNext(result5 -> {
-            assertEquals(result5.getStatusCode(), 204);
-        }).verifyComplete();
+                .assertNext(result5 -> {
+                    assertEquals(result5.getStatusCode(), 204);
+                }).verifyComplete();
     }
 
     @ParameterizedTest
     @MethodSource("com.azure.core.test.TestBase#getHttpClients")
     public void createRoomFullCycleWithOutResponseStep(HttpClient httpClient) {
-        roomsAsyncClient = setupAsyncClient(httpClient, "createRoomFullCycleWithOutResponse");
+        roomsAsyncClient = setupAsyncClient(httpClient,
+                "createRoomFullCycleWithOutResponse");
         assertNotNull(roomsAsyncClient);
-        Mono<CommunicationRoom> response1 = roomsAsyncClient.createRoom(VALID_FROM, VALID_UNTIL, roomJoinPolicy, participants1);
+
+        CreateRoomOptions roomOptions = new CreateRoomOptions()
+                .setValidFrom(VALID_FROM)
+                .setValidUntil(VALID_UNTIL)
+                .setParticipants(participants1);
+
+        Mono<CommunicationRoom> response1 = roomsAsyncClient.createRoom(roomOptions);
 
         StepVerifier.create(response1)
-            .assertNext(roomResult -> {
-                List<RoomParticipant> returnedParticipants = roomResult.getParticipants();
-                assertEquals(participants1.size(), returnedParticipants.size());
-                IntStream.range(0, returnedParticipants.size())
-                    .forEach(x -> assertTrue(areParticipantsEqual(participants1.get(x), returnedParticipants.get(x))));
-            }).verifyComplete();
+                .assertNext(roomResult -> {
+                    assertEquals(true, roomResult.getValidFrom() != null);
+                }).verifyComplete();
 
         String roomId = response1.block().getRoomId();
 
-        Mono<CommunicationRoom> response3 =  roomsAsyncClient.updateRoom(roomId, VALID_FROM, VALID_FROM.plusMonths(3), null, null);
+        UpdateRoomOptions updateOptions = new UpdateRoomOptions()
+                .setValidFrom(VALID_FROM)
+                .setValidUntil(VALID_FROM.plusMonths(3));
+
+        Mono<CommunicationRoom> response3 = roomsAsyncClient.updateRoom(roomId,
+                updateOptions);
 
         StepVerifier.create(response3)
-            .assertNext(result3 -> {
-                assertEquals(result3.getParticipants().size(), 3);
-            }).verifyComplete();
+                .assertNext(result3 -> {
+                    assertEquals(true, result3.getValidUntil().toEpochSecond() > VALID_FROM.toEpochSecond());
```

Comment:
Why just validated validFrrom? We should validate all properties of the result

Reply:
I thought doing an extra check really wouldn't change things since validUntil is only available when the whole room model is returned. But I have now added checks for all room properties.